File: lemur-libs/lemur-contract/src/main/java/io/github/malczuuu/lemur/contract/rest/player/PlayerDto.java
package io.github.malczuuu.lemur.contract.rest.player;

import io.github.malczuuu.lemur.contract.TransportResponse;
import java.time.Instant;
import org.jspecify.annotations.Nullable;

public record PlayerDto(
    String id,
    String name,
    Integer rating,
    String status,
    @Nullable Instant createdDate,
    @Nullable Instant lastModifiedDate,
    Long version)
    implements TransportResponse {}

File: lemur-libs/lemur-contract/src/main/kotlin/io/github/malczuuu/lemur/contract/rest/player/PlayerDto.kt
package io.github.malczuuu.lemur.contract.rest.player

import io.github.malczuuu.lemur.contract.TransportResponse
import kotlin.time.Instant

data class PlayerDto(
    val id: String,
    val name: String,
    val rating: Int,
    val status: String,
    val createdDate: Instant?,
    val lastModifiedDate: Instant?,
    val version: Long,
) : TransportResponse
